openai/openai.go

```diff
@@ -648,6 +628,35 @@ func nameFromToolCallID(messages []Message, toolCallID string) string {
 	return ""
 }
 
+// decodeImageURL decodes a base64 data URI into raw image bytes.
+func decodeImageURL(url string) (api.ImageData, error) {
```

Comment:
More of a side comment but I wonder if we should be doing this validation on the main API side and then bubbling the error down here instead of failing here (since this logic would be duplicated)

Reply:
I think the challenge here is the main API just takes a base64-encoded string, but no mime types (so this is more specific). But I'm not sure why we've been more strict at the compatibility layer anyway, so that's a good question

Reply:
Yeah I think it'll be good to start thinking about which layers we do validation in + surfacing the right errors up. Our validation layers are a bit mixed on compat/native api/cloud